### PR TITLE
HDDS-8109. Seperate Ratis and EC MisReplication Handling

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

Push down the `sendReplicateCommands` from `MisReplicationHandler` into subclasses so that we can have different implementations for Ratis and EC.

Change implementation for Ratis to allow using all sources.

https://issues.apache.org/jira/browse/HDDS-8109

## How was this patch tested?

CI:
https://github.com/adoroszlai/hadoop-ozone/actions/runs/4718760757

### CODE REVIEW

**Comment by sodonnel** [comment]
Looks good apart from the one comment I had above. If you fix that it should be good to commit.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/MisReplicationHandler.java [hunk 1]
```diff
@@ -115,38 +115,11 @@ private Set<ContainerReplica> filterSources(Set<ContainerReplica> replicas) {
         .collect(Collectors.toSet());
   }
 
-  protected abstract ReplicateContainerCommand updateReplicateCommand(
-          ReplicateContainerCommand command, ContainerReplica replica);
-
-  private int sendReplicateCommands(
+  protected abstract int sendReplicateCommands(
```
[the remark this answers is not in the document]
Comment (adoroszlai) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/MisReplicationHandler.java:118]: Thanks, I missed this detail.

**Comment by adoroszlai** [comment]
Looks like `TestRatisMisReplicationHandler` becomes flaky with this change, will fix.

**Comment by adoroszlai** [comment]
[Passed](https://github.com/adoroszlai/hadoop-ozone/actions/runs/4729648618) 100/100 with 4c0d351dd14da1c0ecb473690a1a3201d3818d76.

**Review by sodonnel — APPROVED** [review]

**Comment by adoroszlai** [comment]
Thanks @sodonnel for the review.

**Pull request merged**